sklearn/model_selection/_validation.py

```diff
@@ -488,7 +488,10 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
 
     train_scores = {}
     if parameters is not None:
-        estimator.set_params(**parameters)
+        # clone after setting parameters in case any parameters
+        # are estimators (like pipeline steps)
+        # because pipeline doesn't clone steps in fit
+        estimator = clone(estimator.set_params(**parameters))
```

Comment:
There's a problem if the parameter is assumed to be a *fitted* estimator? But does it ever happen?

Reply:
Could in 3rd party things if a meta-estimator doesn't call fit on an estimator.